app/lib/backend_api_logic/routing_policy.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module BackendApiLogic
+  module RoutingPolicy
+    def policy_chain
+      chain = super
+      return chain unless service.act_as_product?
+      Builder.new(service).to_a.concat(chain)
+    end
+
+    class Builder
+      delegate :proxy, :backend_api_configs, to: :@service
+      delegate :policies_config, to: :proxy
+
+      def initialize(service)
+        @service = service
+      end
+
+      def to_a
+        rules = backend_api_configs.each_with_object([]) do |config, rules|
+          rule = Rule.new(config).as_json
+          rules << rule if rule
+        end
+        return [] if rules.empty?
+        [{
+          name: "routing",
+          version: "builtin",
+          enabled: true,
+          configuration: {
+            operations: rules
+          }
+        }].as_json
+      end
+
+      class Rule
+        delegate :private_endpoint, :path, to: :@config
+
+        def initialize(config)
+          @config = config
+        end
+
+        def as_json
+          return if private_endpoint.blank?
```

Comment:
This is not tested :)
But it makes sense not to route anything if the private_endpoint is not assigned

Reply:
I will test it :)